File: src/main/java/grojdg/harderstats/StatTimer.java
package grojdg.harderstats;

import net.minecraft.util.Util;

public class StatTimer {
    private long timeStarted = 0;
    private boolean isTicking = false;
    private long timeElapsed = 0;

    public void setIsTicking(boolean isTicking) {
        if (!this.isTicking && isTicking) {
            this.timeStarted = Util.getMeasuringTimeMs();
            this.isTicking = true;
        }

        if (this.isTicking && !isTicking) {
            this.timeElapsed = Util.getMeasuringTimeMs() - timeStarted;
            timeStarted = 0;
            this.isTicking = false;
        }
    }

    public long getTimeElapsed() {
        return this.timeElapsed;
    }

    public long getTimeStarted() {
        return this.timeStarted;
    }

    public void reset() {
        this.timeStarted = 0;
        this.isTicking = false;
        this.timeElapsed = 0;
    }
}

File: src/main/kotlin/grojdg/harderstats/StatTimer.kt
package grojdg.harderstats

import net.minecraft.util.Util

class StatTimer {
    var timeStarted: Long = 0
        private set

    var timeElapsed: Long = 0
        private set

    var isTicking = false

    fun startTicking() {
        if (!isTicking) {
            timeStarted = Util.getMeasuringTimeMs()
            isTicking = true
        }
    }

    fun stopTicking() {
        if (isTicking) {
            timeElapsed = Util.getMeasuringTimeMs() - timeStarted
            isTicking = false
        }
    }

    fun reset() {
        timeStarted = 0
        isTicking = false
        timeElapsed = 0
    }
}